src/compas/robots/model/link.py

```diff
@@ -140,74 +137,65 @@ def __init__(self, name, type=None, visual=[], collision=[], inertial=None, **kw
         self.joints = []
         self.parent_joint = None
 
-    def create(self, urdf_importer, meshcls, parent_origin):
+    def create(self, urdf_importer, meshcls, parent_transformation=Transformation()):
         """Recursive function to create all geometry shapes.
         """
         for item in self.visual:
             item.geometry.shape.create(urdf_importer, meshcls)
         for item in self.collision:
             item.geometry.shape.create(urdf_importer, meshcls)
 
-        transformation = Transformation.from_frame(parent_origin)
-
-        # former DAE files have yaxis and zaxis swapped
-        # TODO: already fix in conversion to obj or in import
-        # TODO: move to mesh importer!
-        fx = Frame(parent_origin.point, parent_origin.xaxis, parent_origin.zaxis)
-        transformation_dae = Transformation.from_frame(fx)
-
         for item in self.visual:
-            # set color
             color = item.get_color()
             if color:
                 item.geometry.shape.set_color(color)
-            # transform
-            if type(item.geometry.shape) == MeshDescriptor:
-                if os.path.splitext(item.geometry.shape.filename)[1] == ".dae":
-                    item.geometry.shape.transform(transformation_dae)
-                else:
-                    item.geometry.shape.transform(transformation)
-            else:
-                item.geometry.shape.transform(transformation)
-
+            item.geometry.shape.transform(parent_transformation)
+        
         for item in self.collision:
-            if type(item.geometry.shape) == MeshDescriptor:
-                if os.path.splitext(item.geometry.shape.filename)[1] == ".dae":
-                    item.geometry.shape.transform(transformation_dae)
-                else:
-                    item.geometry.shape.transform(transformation)
-            else:
-                item.geometry.shape.transform(transformation)
+            item.geometry.shape.transform(parent_transformation)
 
         for cjoint in self.joints:
-            cjoint.origin.create(transformation)
+            cjoint.origin.create(parent_transformation)
+            transformation = Transformation.from_frame(cjoint.origin)
             if cjoint.axis:
                 cjoint.axis.create(transformation)
             clink = cjoint.child_link
-            clink.create(urdf_importer, meshcls, cjoint.origin)
+            clink.create(urdf_importer, meshcls, transformation)
 
-    def update(self, joint_state, parent_transformation, reset_transformation):
+    def update(self, joint_state, parent_transformation, reset_transformation, collision=False):
```

Comment:
Shouldn't `collision` default to `True`? Is there really a case where one wants to transform only the visual geometry and not the collision one?

Reply:
you are right!